outlines/models/exllamav2.py

```diff
@@ -162,71 +256,66 @@ def exl2(
+            ExLlamaV2Tokenizer,
         )
+        from exllamav2.generator import ExLlamaV2DynamicGenerator
         from transformers import AutoTokenizer
+
     except ImportError:
         raise ImportError(
-            "The `exllamav2`, `transformers` and `torch` libraries needs to be installed in order to use `exllamav2` models."
+            "The `exllamav2` and `torch` libraries needs to be installed in order to use `exllamav2` models."
         )
+    config = ExLlamaV2Config(model_path)
+    if max_chunk_size is not None:
+        config.max_input_len = max_chunk_size
+        config.max_attention_size = max_chunk_size**2
 
-    # Load tokenizer
-    if not verbose:
-        print(" -- Loading tokenizer...")
-    tokenizer_kwargs.setdefault("padding_side", "left")
-    tokenizer = AutoTokenizer.from_pretrained(model_path, **tokenizer_kwargs)
-    # tokenizer = TransformerTokenizer(model_path, **tokenizer_kwargs)
-
-    # Check fasttensors for config
-    if os.name != "nt":
-        use_fasttensors = True
-    else:
-        use_fasttensors = False
-
-    # Create config
-    config = ExLlamaV2Config()
-    config.model_dir = model_path
-    config.fasttensors = use_fasttensors
-    config.prepare()
-
-    # Set config options
-    if max_seq_len is not None:
-        config.max_seq_len = max_seq_len
-    if scale_pos_emb is not None:
-        config.scale_pos_emb = scale_pos_emb
-    if scale_alpha_value is not None:
-        config.scale_alpha_value = scale_alpha_value
-    if no_flash_attn is not None:
-        config.no_flash_attn = no_flash_attn
-    if num_experts_per_token is not None:
-        config.num_experts_per_token = num_experts_per_token
-    if low_mem:
-        config.set_low_mem()
-
-    # Prepare the model from the config
+    config.arch_compat_overrides()
     model = ExLlamaV2(config)
-
-    # Create cache
-    if cache_8bit:
-        cache = ExLlamaV2Cache_8bit(model, lazy=not model.loaded)
-    elif cache_q4:
-        cache = ExLlamaV2Cache_Q4(model, lazy=not model.loaded)
+    if max_seq_len is None:
+        max_seq_len = -1
+    if cache_q4:
+        cache = ExLlamaV2Cache_Q4(model, max_seq_len=max_seq_len, lazy=True)
     else:
-        cache = ExLlamaV2Cache(model, lazy=not model.loaded)
-
-    # Load the model
-    split = None
-    if gpu_split and gpu_split != "auto":
-        split = [float(alloc) for alloc in gpu_split.split(",")]
-        if not verbose:
-            print(" -- Loading model...")
-        model.load(split)
-
-    # Autoload if no GPU split was provided
-    if not model.loaded:
-        print(" -- Loading model...")
-        model.load_autosplit(cache)
-
-    return ExLlamaV2Model(model, tokenizer, device, cache)
+        cache = ExLlamaV2Cache(model, max_seq_len=max_seq_len, lazy=True)
+    model.load_autosplit(cache, progress=True)
+
+    print("Loading tokenizer...")
+    tokenizer = ExLlamaV2Tokenizer(config)
+    tokenizer.vocabulary = tokenizer.extended_piece_to_id
```

Comment:
Could you implement a function which converts an EXL2 tokenizer to a `TransformerTokenizer` rather than patching here?

It would ensure downstream users don't need to define their own function which applies the same logic.

Reply:
@lapp0 oh do you mean instead of having the hf_tokenizer like below create the hf_tokenizer from the exllamav2's tokenizer using a function?

Reply:
Exactly. This way downstream users won't have to define their own `patch_tokenizer()` to use our logits processors as I do currently in the example here:

https://github.com/lapp0/exllamav2/pull/1/files#diff-26f303de07c10aad998e33d3df52581643673a598162cc4b35ef051f52d7c60bR68-R81

Reply:
Done!